managers/backup-manager/BackupManager.js

```diff
@@ -0,0 +1,421 @@
+                logs: logs,
+                snapshotId: lastOperation.snapshotId
+              })
+            ).then(() => this.backupStore.getBackupFile(options))
+            .then(metadata => eventmesh.apiServerClient.updateOperationResponse({
+              resourceId: options.instance_guid,
+              operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+              operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+              operationId: metadata.backup_guid,
+              value: metadata
+            }));
+        }
+      });
+  }
+
+  deleteBackup(options) {
+    logger.info('Attempting delete with:', options);
+    return this.backupStore
+      .deleteBackupFile(options)
+      .then(() => eventmesh.apiServerClient.updateOperationState({
+        resourceId: options.instance_guid,
+        operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+        operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+        operationId: options.backup_guid,
+        stateValue: CONST.APISERVER.RESOURCE_STATE.DELETED
+      }))
+      .catch(err => {
+        return Promise
+          .try(() => logger.error(`Error during delete of backup`, err))
+          .tap(() => eventmesh.apiServerClient.updateOperationState({
+            resourceId: options.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: options.backup_guid,
+            stateValue: CONST.APISERVER.RESOURCE_STATE.ERROR
+          }))
+          .tap(() => eventmesh.apiServerClient.updateOperationResponse({
+            resourceId: options.instance_guid,
+            operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+            operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+            operationId: options.guid,
+            value: err
+          }));
+      });
+  }
+
+  abortLastBackup(abortOptions, force) {
+    logger.info('Starting abort with following options:', abortOptions);
+    return eventmesh.apiServerClient.updateOperationState({
+      resourceId: abortOptions.instance_guid,
+      operationName: CONST.APISERVER.ANNOTATION_NAMES.BACKUP,
+      operationType: CONST.APISERVER.ANNOTATION_TYPES.BACKUP,
+      operationId: abortOptions.guid,
+      stateValue: CONST.OPERATION.ABORTING
+    }).then(() => this.backupStore
+      .getBackupFile({
+        tenant_id: this.getTenantGuid(abortOptions.context),
+        service_id: abortOptions.service_id,
+        instance_guid: abortOptions.instance_guid
+      })).then(metadata => {
+      if (!force && metadata.trigger === CONST.BACKUP.TRIGGER.SCHEDULED) {
+        throw new Forbidden('System scheduled backup runs cannot be aborted');
+      }
+      switch (metadata.state) {
+      case 'processing':
+        return this.agent
+          .abortBackup(metadata.agent_ip)
+          .return({
+            state: CONST.OPERATION.ABORTING
```

Comment:
Dont you want to push the lines 384-390, here at 402?

Reply:
Done